src/contracts/payloads/WithPayloadsManager.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IWithPayloadsManager} from './interfaces/IWithPayloadsManager.sol';
+import {OwnableWithGuardian} from 'solidity-utils/contracts/access-control/OwnableWithGuardian.sol';
+import {Errors} from '../libraries/Errors.sol';
+
+/**
+ * @title WithPayloadsManager
+ * @author BGD Labs
+ * @dev Contract module which provides a basic access control mechanism, where
+ * there are accounts (owner, guardian, and payloads manager) which can be granted
+ * exclusive access to specific functions.
+ * @notice By default, all the roles will be assigned to the one that deploys the contract. This
+ * can later be changed with appropriate functions.
+ */
+contract WithPayloadsManager is OwnableWithGuardian, IWithPayloadsManager {
+  address private _payloadsManager;
+
+  constructor() {
```

Comment:
would it make sense to add here address for guardian / owner? instead of using message sender, and then having to update to the correct addresses?

Reply:
I made it in the same way as OwnableWithGuardian. Is it a bad approach?

Reply:
This case is only for PayloadsController, so not really the same as OwnableWithGuardian which is generic. and here we are even inheriting. I think it could make sense to deploy already with the owner / guardian (would simplify deployment). But was just a suggestion. Not full sure if it makes sense or not (not supper important either). 
Just opening the discussion

Reply:
I mean anyway it's deployed as proxy. So owner, payloads manager, guardian will be initialized with `initialize` function of contract. I don't really understand what's a proposal